client/js/widgets/button.js

```diff
@@ -44,22 +44,186 @@ export class Button extends Widget {
   applyVariables(field, variables, problems) {
     if(Array.isArray(field.applyVariables)) {
       for(const v of field.applyVariables) {
-        if(v.parameter && v.variable) {
-          field[v.parameter] = (v.index === undefined) ? variables[v.variable] : variables[v.variable][v.index];
+        if (v.var_operation)
+          v.operation = variables[v.var_operation];
+        if (v.operation) {
+          let op = [null, 1, 1, 1];
+          Object.entries(v).forEach(([key, value]) => {
+            op[key[key.length-1]] = value;
+            if (['var_operand1', 'var_operand2', 'var_operand3'].indexOf(key) >= 0)
+              op[key[key.length-1]] = variables[value].index !== undefined ? variables[value][value.index] : variables[value];
+          });
+          variables['TEMP'] = this.compute(v.operation, v.variable, op[1], op[2], op[3],problems);
```

Comment:
Maybe call it LAST or PREVIOUS or something like that?

It's probably breaking if the variables for the operands didn't exist.

Reply:
Yeah, I had considered a bunch of different names, like LAST, PREVIOUS, CURRENT, INTERIM, RESULT and ended up with TEMP because it better conveys the idea that this variable is meant to be used only transitory during the applyVariables scope. I'm ok with changing the name to whatever people feel most adequate.